server/src/main/java/io/crate/execution/engine/collect/sources/ShardCollectSource.java

```diff
@@ -491,6 +491,8 @@ private ShardCollectorProvider getCollectorProviderSafe(ShardId shardId) {
                     shardRowContexts.add(shardCollectorProvider.shardRowContext());
                 } catch (ShardNotFoundException | IllegalIndexShardStateException e) {
                     unassignedShards.add(toUnassignedShard(index.getName(), shard.value));
+                } catch (Throwable t) {
+                    LOGGER.error("Error retrieving shard information for index={} error={}", index.getName(), t);
```

Comment:
Maybe we could add an additional shard state `corrupted` and add the error message to it. So users can still see the shards for tables that have a corrupt schema?

Reply:
> Maybe we could add an additional shard state corrupted and add the error message to it. So users can still see the shards for tables that have a corrupt schema?

Good idea, that would at least give us hints for potential masked bugs here. +1

Reply:
+1